### PR TITLE
(maint) update apache httpcore and httpasyncclient

### PR DESCRIPTION
**Pull request by jonathannewman**
(maint) update apache httpcore and httpasyncclient

This updates the apache httpcore library to 4.4.15 to address a
failure seen where requests were blocked at the client level
when testing TLS 1.3.  This also updates the httpasyncclient
to the most recent version.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 ## [Unreleased]
+- update httpasync-client to 4.1.5 to fix async issues seen in testing.  https://downloads.apache.org/httpcomponents/httpasyncclient/RELEASE_NOTES-4.1.x.txt
+- update httpasync-client to 4.4.15 to fix async issues seen in testing.  https://downloads.apache.org/httpcomponents/httpcore/RELEASE_NOTES-4.4.x.txt (HTTPCORE-687 and "Bug fix: Non-blocking TLSv1.3 connections can end up in an infinite event spin when closed
+  concurrently by the local and the remote endpoints." 
+- update logback to 1.2.9 to address https://nvd.nist.gov/vuln/detail/CVE-2021-42550
```
Comment (Magisus) [inline CHANGELOG.md:5]: This is also in Nick's PR, maybe we only need one of them?

**Review by mschmele — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
 ## [Unreleased]
+- update httpasync-client to 4.1.5 to fix async issues seen in testing.  https://downloads.apache.org/httpcomponents/httpasyncclient/RELEASE_NOTES-4.1.x.txt
+- update httpasync-client to 4.4.15 to fix async issues seen in testing.  https://downloads.apache.org/httpcomponents/httpcore/RELEASE_NOTES-4.4.x.txt (HTTPCORE-687 and "Bug fix: Non-blocking TLSv1.3 connections can end up in an infinite event spin when closed
+  concurrently by the local and the remote endpoints."
```
Comment (mwaggett) [inline CHANGELOG.md:4]: missing a closing parenthesis somewhere in here

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,8 @@
 ## [Unreleased]
+- update httpasync-client to 4.1.5 to fix async issues seen in testing.  https://downloads.apache.org/httpcomponents/httpasyncclient/RELEASE_NOTES-4.1.x.txt
+- update httpasync-client to 4.4.15 to fix async issues seen in testing.  https://downloads.apache.org/httpcomponents/httpcore/RELEASE_NOTES-4.4.x.txt (HTTPCORE-687 and "Bug fix: Non-blocking TLSv1.3 connections can end up in an infinite event spin when closed
```
Comment (mwaggett) [inline CHANGELOG.md:3]:
```suggestion
- update httpasync-core to 4.4.15 to fix async issues seen in testing.  https://downloads.apache.org/httpcomponents/httpcore/RELEASE_NOTES-4.4.x.txt (HTTPCORE-687 and "Bug fix: Non-blocking TLSv1.3 connections can end up in an infinite event spin when closed
```

**Review by nmburgan — APPROVED** [review]
I can rebase mine and move the changelog into the new version on my PR once this is merged.

**Review by donoghuc — COMMENTED** [review]
Ran orch tests with these versions locally. 👍

**Comment by Magisus** [comment]
👍 Puppet server is green

**Review by Magisus — APPROVED** [review]

**Comment by nmburgan** [comment]
Probably want to do a clj-parent release with the logback and jackson databind fixes for this next release.  We'll bump the TLSv1.3 stuff to the following release.

**Pull request merged**